crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1363,25 +1415,89 @@ impl SymbolicationActor {
+            );
+            metric!(
+                time_raw("symbolication.unparsable_modules") = unparsable_modules,
+                "platform" => &platform, "origin" => &origin,
             );
 
-            metric!(time_raw("symbolication.num_stacktraces") = stacktraces.len() as u64);
-            metric!(time_raw("symbolication.truncated_stacktraces") = metrics.truncated_traces);
-            metric!(time_raw("symbolication.bad_stacktraces") = metrics.bad_traces);
+            metric!(
+                time_raw("symbolication.num_stacktraces") = stacktraces.len() as u64,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.short_stacktraces") = metrics.short_traces,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.truncated_stacktraces") = metrics.truncated_traces,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.bad_stacktraces") = metrics.bad_traces,
+                "platform" => &platform, "origin" => &origin,
+            );
 
             metric!(
                 time_raw("symbolication.num_frames") =
-                    stacktraces.iter().map(|s| s.frames.len() as u64).sum()
+                    stacktraces.iter().map(|s| s.frames.len() as u64).sum(),
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.scanned_frames") = metrics.scanned_frames,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.unsymbolicated_frames") = metrics.unsymbolicated_frames,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.unsymbolicated_context_frames") =
+                    metrics.unsymbolicated_context_frames,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.unsymbolicated_cfi_frames") =
+                    metrics.unsymbolicated_cfi_frames,
+                "platform" => &platform, "origin" => &origin,
+            );
+            metric!(
+                time_raw("symbolication.unsymbolicated_scanned_frames") =
+                    metrics.unsymbolicated_scanned_frames,
+                "platform" => &platform, "origin" => &origin,
             );
-            metric!(time_raw("symbolication.scanned_frames") = metrics.scanned_frames);
             metric!(
-                time_raw("symbolication.unsymbolicated_frames") = metrics.unsymbolicated_frames
+                time_raw("symbolication.unmapped_frames") = metrics.unmapped_frames,
+                "platform" => &platform, "origin" => &origin,
```

Comment:
This function already mixes runtime maintenance (creating futures and spawning them in threadpools in a specific order and nesting) and business logic.  Adding all these metrics makes it even harder to follow what is going on.
Would it be possible to move all the metrics into a `record_symbolication_metrics()` method?  Or would the signature of such a thing be unreadable as well?